primitives/types/transaction_source.go

```diff
@@ -0,0 +1,68 @@
+package types
+
+import (
+	"bytes"
+
+	sc "github.com/LimeChain/goscale"
+)
+
+// The source of the transaction.
+//
+// Depending on the source we might apply different validation schemes.
+// For instance we can disallow specific kinds of transactions if they were not produced
+// by our local node (for instance off-chain workers).
+const (
+	// Transaction is already included in block.
+	//
+	// This means that we can't really tell where the transaction is coming from,
+	// since it's already in the received block. Note that the custom validation logic
+	// using either `Local` or `External` should most likely just allow `InBlock`
+	// transactions as well.
+	InBlock = TransactionSourceValue(iota)
+
+	// Transaction is coming from a local source.
+	//
+	// This means that the transaction was produced internally by the node
+	// (for instance an Off-Chain Worker, or an Off-Chain Call), as opposed
+	// to being received over the network.
+	Local
+
+	// Transaction has been received externally.
+	//
+	// This means the transaction has been received from (usually) "untrusted" source,
+	// for instance received over the network or RPC.
+	External
+)
+
+type TransactionSourceValue sc.U8
+
+func (ts TransactionSourceValue) Encode(buffer *bytes.Buffer) {
+	sc.U8(ts).Encode(buffer)
+}
+
+func DecodeTransactionSourceValue(buffer *bytes.Buffer) TransactionSourceValue {
+	return TransactionSourceValue(sc.DecodeU8(buffer))
+}
+
+func (ts TransactionSourceValue) Bytes() []byte {
+	return sc.EncodedBytes(ts)
+}
+
+type TransactionSource sc.VaryingData
```

Comment:
Given that this is an enum, shouldn't it be a SCALE uint type?

Reply:
the underlying type is still U8, but this way it gives me compile type type errors

`txSource := types.NewTransactionSource(types.External) // (2)`

if I try 
`txSource := types.NewTransactionSource(sc.U8(5))` which is invalid